aeppl/censoring.py

```diff
@@ -0,0 +1,214 @@
+from typing import List, Optional
+
+import aesara.tensor as at
+import numpy as np
+from aesara.graph.basic import Node
+from aesara.graph.fg import FunctionGraph
+from aesara.graph.rewriting.basic import node_rewriter
+from aesara.scalar.basic import Ceil, Clip, Floor, RoundHalfToEven
+from aesara.scalar.basic import clip as scalar_clip
+from aesara.tensor.elemwise import Elemwise
+from aesara.tensor.var import TensorConstant
+
+from aeppl.abstract import MeasurableVariable, assign_custom_measurable_outputs
+from aeppl.logprob import CheckParameterValue, _logcdf, _logprob, logdiffexp
+from aeppl.rewriting import measurable_ir_rewrites_db
+
+
+class MeasurableClip(Elemwise):
+    """A placeholder used to specify a log-likelihood for a clipped RV sub-graph."""
```

Comment:
N.B. It would be much better if we could assign a custom IR class/type for the combination of `Elemwise` and `Clip` and not just the `Elemwise`.  Perhaps we could use a general `MeasurableElemwise` instead, because the name `MeasurableClip` is misleading when `Clip` is not involved at the type/class-level, nor is use of a scalar `Clip` `Op` enforced at run-time.

Reply:
Added a runtime check in the `clip_logprob`

Reply:
A run-time check can't really address this kind of type-level ambiguity.  Regardless, a sufficient run-time check would need to take place in this type's constructor, so that `MeasurableClip`s that don't use a scalar `Clip` `Op` can't be constructed.

Reply:
Added a` MeasurableElemwise` that checks for compatible scalar_op types at creation. What do you think?